cmd/horcrux/cmd/state.go

```diff
@@ -5,11 +5,12 @@ import (
 	"errors"
 	"fmt"
 	"os"
-	"os/exec"
 	"path"
+	"path/filepath"
 	"strconv"
 
 	"github.com/mitchellh/go-homedir"
+	"github.com/shirou/gopsutil/process"
```

Comment:
gopsutil is on v3 now, so -- if going forward with gopsutil -- the import path should be `github.com/shirou/gopsutil/v3/process`, and go.mod and go.sum will need updated.

Reply:
`pgrep` is probably still fine. `pgrep -d, -x horcrux` would return a list like `14094,23680,80341`, and then you can decide how to process that in Go, whether you just `strings.Count(output, ",")` or actually extract the PIDs and filter away `os.Getpid()`.

That assumes there is no false positive on `pgrep horcrux` and nobody is doing anything funny like renaming the horcrux binary, or using exec and changing `argv[0]`.

The lockfile approach seems a little safer in those regards, as it is much less likely that a rogue file would put something in e.g. `~/.horcrux/horcrux.lock`.

Reply:
I see what you mean. The lockfile will be nice too because then you can run multiple horcrux instances out of different config directories and these commands will work properly based on the horcrux runtime state of the applicable config directory.